src/kademlia/node.cpp

```diff
@@ -1191,28 +1190,44 @@ void node::write_nodes_entries(sha1_hash const& info_hash
 		if (wanted_node == m_nodes.end()) continue;
 		std::vector<node_entry> n;
 		wanted_node->second->m_table.find_node(info_hash, n, 0);
-		r[wanted_node->second->protocol_nodes_key()] = write_nodes_entry(n);
+		r[wanted_node->second->protocol().nodes_key()] = write_nodes_entry(n);
 	}
 }
 
-node::protocol_descriptor const& node::map_protocol_to_descriptor(udp protocol)
+udp protocol_descriptor::protocol() const
 {
-	static protocol_descriptor descriptors[] =
-		{ {udp::v4(), "n4", "nodes"}
-		, {udp::v6(), "n6", "nodes6"} };
+	return m_protocol;
+}
 
-	for (int i = 0; i < sizeof(descriptors) / sizeof(protocol_descriptor); ++i)
-	{
-		if (descriptors[i].protocol == protocol)
-			return descriptors[i];
-	}
+char const* protocol_descriptor::family_name() const
+{
+	return is_v4() ? "n4" : "n6";
```

Comment:
good idea, I will add a private inner structure to `protocol_descriptor` with that trick